upcloud/managed_database.go

```diff
@@ -0,0 +1,693 @@
+	// native API.
+	ManagedDatabaseUserTypeNormal = "normal"
+)
+
+// Authentication types
+const (
+	// ManagedDatabaseUserAuthenticationCachingSHA2Password selects "caching_sha2_password" type of authentication type.
+	// This type is only supported with MySQL services.
+	//nolint:gosec
+	ManagedDatabaseUserAuthenticationCachingSHA2Password = "caching_sha2_password"
+	// ManagedDatabaseUserAuthenticationMySQLNativePassword selects "mysql_native_password" type of authentication type.
+	// This type is only supported with MySQL services.
+	//nolint:gosec
+	ManagedDatabaseUserAuthenticationMySQLNativePassword = "mysql_native_password"
+)
+
+// ManagedDatabase represents an existing managed database instance
+type ManagedDatabase struct {
+	Backups          []ManagedDatabaseBackup         `json:"backups"`
+	Components       []ManagedDatabaseComponent      `json:"components"`
+	CreateTime       time.Time                       `json:"create_time"`
+	Maintenance      ManagedDatabaseMaintenance      `json:"maintenance"`
+	Name             string                          `json:"name"`
+	NodeCount        int                             `json:"node_count"`
+	NodeStates       []ManagedDatabaseNodeState      `json:"node_states"`
+	Plan             string                          `json:"plan"`
+	Powered          bool                            `json:"powered"`
+	Properties       ManagedDatabaseProperties       `json:"properties"`
+	State            string                          `json:"state"`
+	Title            string                          `json:"title"`
+	Type             string                          `json:"type"`
+	UpdateTime       time.Time                       `json:"update_time"`
+	ServiceURI       string                          `json:"service_uri"`
+	ServiceURIParams ManagedDatabaseServiceURIParams `json:"service_uri_params"`
+	Users            []ManagedDatabaseUser           `json:"users"`
+	UUID             string                          `json:"uuid"`
+	Zone             string                          `json:"zone"`
+}
+
+// ManagedDatabaseBackup represents a taken full backup's point in time
+type ManagedDatabaseBackup struct {
+	BackupTime time.Time `json:"backup_time"`
+	DataSize   int       `json:"data_size"`
+}
+
+// ManagedDatabaseComponent represents an accessible component within a service. The usage varies between service types
+type ManagedDatabaseComponent struct {
+	Component string `json:"component"`
+	Host      string `json:"host"`
+	Port      int    `json:"port"`
+	// Route describes how the component can be reached.
+	// 	public  = The component can be reached over public internet
+	// 	dynamic = The component can be only reached over utility network
+	Route string `json:"route"`
+	// Usage describes the role of the component.
+	// 	primary = The component is a primary (writable) instance in a cluster
+	// r	eplica = The component is a standby (read-only) instance in a cluster
+	Usage string `json:"usage"`
```

Comment:
Should these be made into consts / custom types / Go-style enums, if these are the only options (currently) available?

Reply:
yes, though they are never used in requests so I'll get warnings of unused exported constants :)

Reply:
Well, I guess one could add some references in tests for them, giving IDEA (I'm guessing? Sometimes the warnings are not that great..) some ideas?

Reply:
I will look into these and maybe add some dummy validation in the integration tests.

Reply:
implemented